docs/uploader/34_development.md

```diff
@@ -0,0 +1,44 @@
+
+### Overview of components
+
+#### The [oam-uploader-api](https://github.com/hotosm/oam-uploader-api):
+The API provides the endpoints used by the upload submission form, including routes for authenticating the session via a database of approved tokens, adding image URLs and metadata to the processing queue database, checking the status of a user's uploads, and checking the status of specific uploads. It also provides the API endpoints used for managing users through the administrative form, including adding, removing and editing tokens. Finally, it manages a queue of images by checking for new submissions to the URL database at regular intervals, downloading them, processsing them to a standardized TIFF specification, generating metadata and thumbnails, and uploading them to an Amazon S3 bucket. The other two components require a connection to the API to function.
+
+#### The [oam-uploader-admin](https://github.com/hotosm/oam-uploader):
+The administrative panel uses API calls to add, remove, and edit authenticated tokens, which substitute for users in the upload system. The tokens themselves are random strings that are generated by the API when a PUT request is made to the tokens route. In development, the tokens database is temporary, so it is necessary to serve the admin panel in addition to the API in order to populate the system with a valid token.
+
+#### The [oam-uploader](https://github.com/hotosm/oam-uploader):
+The repo referred to as the uploader contains the front-end form which allows authenticated users to publish image URLs to the processing queue, and to check the processing status of submitted tasks. It depends on a connection to the API in order to submit datasets and associated metadata to the processing queue, check upload status, and authenticate tokens. As described above, this page depends on the API for most of its functionality, and depends on the admin panel to trigger token generation, so both components must be active during development.
+
+### Other development considerations
+
+- It will be necessary to use [nvm](https://github.com/creationix/nvm) in development- the two forms must be built use node 4.x, while the API must be built using node 0.12.
+
+- When launching the API, several environment variables must be set using the following command:
+
+  ```
+  GDAL_TRANSLATE_BIN={your GDAL 2 path} ADMIN_USERNAME={your choice} ADMIN_PASSWORD={your choice} AWS_SECRET_KEY_ID={} AWS_SECRET_ACCESS_KEY={} AWS_REGION=us-east-1 OIN_BUCKET=hotosm-oam-staging DBURI={} DBURI_TEST=mongodb://localhost/oam-uploader-test node index.js
+  ```
+
+  The script does not fully inherit the variables in the local.env file, so this override is necessary even if it has been populated. Contact a maintainer for the full AWS credentials and database paths.
```

Comment:
You can do `source local.env` before starting the api to load all the variables.
Also, variables set from ENV, will always override the ones in the config files.